chapters/06_agent_based_extension/05_reinforcement_learning/main.tex

```diff
@@ -0,0 +1,478 @@
+\section{Reinforcement learning algorithm}\label{sec:reinforcement_learning}
+
+The agent based mode described in Section~\ref{sec:agent_based_model} describes
+how the agents can interact with the environment.
+The next step is to describe how the agents can learn from their interactions.
+A reinforcement learning algorithm is described in this section where the agents
+update their service rates based on the utilities they receive from the
+queueing system.
+The concepts described in Section~\ref{sec:queueing_section} and
+Section~\ref{sec:agent_based_model} are incorporated in the reinforcement
+learning algorithm so that the agents decide how fast they should serve the
+customers in order to maximise their own utility.
+
+The reinforcement learning algorithm is a policy iteration algorithm where
+the agents update their service rates based on the utilities they receive
+from the queueing system.
+The service rates will be referred to as the policy in this section.
+A policy is a set of service rates for every server for each possible state
+they can be in.
```

Comment:
Need a couple of text book references peppered throughout here.

Reply:
Added ✔️